akka-stream/src/main/scala/akka/stream/actor/ActorPublisher.scala

```diff
@@ -52,7 +52,7 @@ object ActorPublisherMessage {
    * subscription.
    */
   final case object Cancel extends Cancel with NoSerializationVerificationNeeded
-  sealed class Cancel extends ActorPublisherMessage
+  sealed abstract class Cancel extends ActorPublisherMessage
 
```

Comment:
it looks like we are missing the Java API for `SubscriptionTimeoutExceeded`. Would you mind adding that also?
It should be similar to the below `cancelInstance`.
Please also move those close together with classes/objects.